include/rdma/fi_eq.h

```diff
@@ -73,9 +71,12 @@ struct fid_wait {
 	struct fi_ops_wait	*ops;
 };
 
+struct pthread_mutex_t;
+struct pthread_cond_t;
```

Comment:
Don't do this.  The `struct pthread_XXX_t` form is **nonstandard** and we should not be messing with their namespace.  Leave the `#include <pthread.h>` and guard it with an `#if !WINDOWS`, [per Sean's comment earlier](https://github.com/ofiwg/libfabric/pull/2041#discussion_r63965793).

Similarly, put an `#if !WINDOWS` around the members of the `struct fi_mutex_cond` below.  When it comes time to add Windows support properly, we can look at whether to put Windows style mutexes here with an #ifdef or defining an additional comparable type for the Windows platform (`struct fi_win_mutex_cond` or whatever).